pkg/logger/logger.go

```diff
@@ -64,90 +154,179 @@ func ParseLogLevel(logLevel string) (LogLevel, error) {
 	return "", fmt.Errorf("Invalid log level `%s`. Valid options are: %v", logLevel, validLevels)
 }
 
-func (l *Logger) log(logColor *color.Color, message string) {
-	if l.File != "" {
-		if l.File == "/dev/stdout" {
-			_, err := logColor.Fprintln(os.Stdout, message)
-			if err != nil {
-				color.Red("%s\n", err)
-			}
-		} else if l.File == "/dev/stderr" {
-			_, err := logColor.Fprintln(os.Stderr, message)
-			if err != nil {
-				color.Red("%s\n", err)
-			}
-		} else {
-			f, err := os.OpenFile(l.File, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0o644)
-			if err != nil {
-				color.Red("%s\n", err)
-				return
-			}
-
-			defer func(f *os.File) {
-				err = f.Close()
-				if err != nil {
-					color.Red("%s\n", err)
-				}
-			}(f)
+// getLogOutputWriter returns the appropriate writer for logging based on the log file path.
+// This is a helper for the log method to handle different output destinations.
+func (l *Logger) getLogOutputWriter() (io.Writer, func(), error) {
+	// Default to stdout
+	if l.File == "" {
+		return os.Stdout, func() {}, nil
+	}
 
-			_, err = f.Write([]byte(fmt.Sprintf("%s\n", message)))
+	switch l.File {
+	case "/dev/stdout":
+		return os.Stdout, func() {}, nil
+	case "/dev/stderr":
+		return os.Stderr, func() {}, nil
+	case "/dev/null":
+		return io.Discard, func() {}, nil
+	default:
+		// For actual files, open the file
+		f, err := os.OpenFile(l.File, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0o644)
+		if err != nil {
+			return nil, nil, err
+		}
+		
+		// Return a cleanup function to close the file
+		cleanup := func() {
+			err := f.Close()
 			if err != nil {
 				color.Red("%s\n", err)
 			}
 		}
+		
+		return f, cleanup, nil
+	}
+}
+
+func (l *Logger) log(logColor *color.Color, message string) {
```

Comment:
Where do we use this? This is wrong to me because it's different from the charmbracelet log function, which accepts as the first arg the log level.